src/createAuthContainer.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import * as firebase from 'firebase';
+
+import * as utils from './utils';
+
+const createAuthContainer = (WrappedComponent, userPropName = 'user') => {
+  if (!utils.isReactComponent(WrappedComponent)) {
+    throw new Error('First argument to createAuthContainer() must be a React component.');
+  } else if (typeof userPropName !== 'string' || userPropName === '') {
+    throw new Error('Second argument to createAuthContainer() must be a non-empty string representing a prop name.');
+  }
+
+  let unsubscribe;
+
+  class CreateAuthContainer extends React.Component {
```

Comment:
I'm conflicted and honestly not sure what the most correct thing to call this is. Let's chat about it in person tomorrow.